src/Pages/Main/BannerSlider/BannerSlider.js

```diff
@@ -0,0 +1,395 @@
+    Funtion 컴포넌트로 변경 후 setInterval 고장남.
+
+    2) next, prev Button Control 
+    현재 progressWidth 초기화 후 다음 또는 이전 배너로 슬라이드
+
+    3) play, pause Play Control
+    pause : progressWidth 유지하고 setInterval clear
+    play : setInterval 시작
+*/
+
+function BannerSlider() {
+  const [slideIndex, setSlideIndex] = useState(0);
+  const [progressWidth, setProgressWidth] = useState(0);
+  const [isRunning, setIsRunning] = useState(true);
+  const [slider, setSlider] = useState({});
+
+  // auto play time (5초)을 프로그래스바로 보여주기 위한 Interval
+  useInterval(
+    () => {
+      if (progressWidth >= 100) {
+        setProgressWidth(0);
+        next();
+      } else {
+        setProgressWidth(progressWidth + 1 / (0.1 + 5));
+      }
+    },
+    isRunning ? 10 : null
+  );
+
+  // Next 슬라이드 Event
+  const next = () => {
+    setSlideIndex(slideIndex + 1 >= BANNERS.length ? 0 : slideIndex + 1);
+    setProgressWidth(0);
+    slider.slickNext();
+  };
+
+  // Prev 슬라이드 Event
+  const previous = () => {
+    setSlideIndex(slideIndex === 0 ? BANNERS.length - 1 : slideIndex - 1);
+    slider.slickPrev();
+  };
+
+  // ProgressBar Play Event
+  const play = () => {
+    setIsRunning(!isRunning);
+  };
+
+  // ProgressBar Pause Event
+  const pause = () => {
+    setIsRunning(!isRunning);
+  };
+
+  const settings = {
+    dots: true,
+    infinite: true,
+    speed: 500,
+    slidesToShow: 1,
+    slidesToScroll: 1,
+    arrows: false,
+  };
+
+  return (
+    <PromotionBanner>
+      <BannerBg color={BANNERS[slideIndex].color}>
+        <BannerInner>
+          <BannerItem>
+            <BannerInfo href={BANNERS[slideIndex].link} target="_blank">
+              <div className="title">
+                {BANNERS[slideIndex].title.split('\n').map((line, i) => (
+                  <div key={i}>
+                    {line}
+                    <br />
+                  </div>
+                ))}
+              </div>
+              <a
+                href={BANNERS[slideIndex].link}
+                target="_blank"
+                rel="noreferrer"
+              >
+                자세히 보기
+              </a>
+            </BannerInfo>
+            <ProgressBarArea>
+              <div className="progressBar">
+                <div
+                  className="progress"
+                  style={{ width: `${progressWidth}%` }}
+                ></div>
+              </div>
+              <span className="current">
+                {('00' + (slideIndex + 1)).slice(-2)}
+              </span>
+              <span className="total">{('00' + BANNERS.length).slice(-2)}</span>
+              <PlayControlArea onClick={isRunning ? pause : play}>
+                {isRunning ? (
+                  <i class="fas fa-pause"></i>
+                ) : (
+                  <i class="fas fa-play"></i>
+                )}
+              </PlayControlArea>
+            </ProgressBarArea>
+            <BannerWrapper>
+              <Slider ref={c => setSlider(c)} {...settings}>
```

Comment:
- ref 를 사용하시려면 useRef 를 사용해주세요!
```suggestion
function BannerSlider() {
  const slider = useRef();  
  ...
              <Slider ref={slider} {...settings}>
```

Reply:
useRef 사용하였습니다! 😃